pkg/controller/controller.go

```diff
@@ -0,0 +1,272 @@
+package controller
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/submariner-io/admiral/pkg/federate"
+	mcservice "github.com/submariner-io/lighthouse/pkg/apis/lighthouse.submariner.io/v1"
+	core_v1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+)
+
+type remoteServiceMap struct {
+	remoteServices map[string]*mcservice.MultiClusterService
+}
+
+type LightHouseController struct {
+	//remoteServices is a map that holds the remote services that are discovered.
+	remoteServiceMap remoteServiceMap
+
+	// remoteClusters is a map of remote clusters, which have been discovered
+	remoteClusters map[string]*RemoteCluster
+
+	processingMutex *sync.Mutex
+
+	federator federate.Federator
+
+	// Indirection hook for unit tests to supply fake client sets
+	newClientset func(kubeConfig *rest.Config) (kubernetes.Interface, error)
+}
+
+type RemoteCluster struct {
+	stopCh          chan struct{}
+	ClusterID       string
+	serviceInformer cache.SharedIndexInformer
+	queue           workqueue.RateLimitingInterface
+}
+
+func New(federator federate.Federator) *LightHouseController {
+	return &LightHouseController{
+		federator:        federator,
+		remoteServiceMap: remoteServiceMap{make(map[string]*mcservice.MultiClusterService)},
```

Comment:
create a type for `map[string]*mcservice.MultiClusterService` to make the code more maintainable.

Reply:
same consideration goes for `RemoteCluster`